examples/platform/silabs/SiWx917/SiWx917DeviceDataProvider.cpp

```diff
@@ -16,20 +16,158 @@
  */
 
 #include "SiWx917DeviceDataProvider.h"
+#include "DeviceConfig.h"
+#include "siwx917_utils.h"
 #include <crypto/CHIPCryptoPAL.h>
 #include <lib/support/Base64.h>
 #include <platform/silabs/SilabsConfig.h>
+#include <setup_payload/Base38Decode.h>
 #include <setup_payload/Base38Encode.h>
+#include <setup_payload/QRCodeSetupPayloadGenerator.h>
 #include <setup_payload/SetupPayload.h>
 
 namespace chip {
 namespace DeviceLayer {
-namespace EFR32 {
-
+namespace SIWx917 {
+#include <string>
 // using namespace chip::Credentials;
 using namespace chip::DeviceLayer::Internal;
 
-CHIP_ERROR EFR32DeviceDataProvider::GetSetupDiscriminator(uint16_t & setupDiscriminator)
+#ifdef SIWX917_USE_COMISSIONABLE_DATA
+void SIWx917DeviceDataProvider::setupPayload(uint8_t * outBuf)
+{
+    SetupPayload payload;
+    std::string result;
+    ChipError err   = CHIP_NO_ERROR;
+    payload.version = 0;
+    payload.discriminator.SetLongValue(discriminatorValue);
+    payload.setUpPINCode = passcode;
+    payload.rendezvousInformation.SetValue((chip::RendezvousInformationFlags) rendezvousFlag);
+    payload.commissioningFlow = (chip::CommissioningFlow) commissionableFlow;
+    payload.vendorID          = vendorId;
+    payload.productID         = productId;
+
+    QRCodeSetupPayloadGenerator generator(payload);
+    err = generator.payloadBase38Representation(result);
+    if (err != CHIP_NO_ERROR)
+    {
+        SILABS_LOG("Failed to get the payload: %d", err);
+    }
+    SILABS_LOG("Payload value in string format : %s", result.c_str());
+    std::vector<uint8_t> result1;
+    // skipping the MT: from the payload during decoding
+    err = base38Decode(result.substr(3), result1);
+    if (err == CHIP_NO_ERROR)
+    {
+        for (uint8_t i = 0; i < result1.size(); i++)
+        {
+            outBuf[i] = result1.at(i);
```

Comment:
The lack of a length check here is pretty fragile, but ok.

Reply:
Resolving to merge - platform owner approved this potential out of bounds access :/

Reply:
Merged, now re-opening for visibility. The method should have taken a span as input to do bounds checking.